### ISSUE TITLE
Add Graph API version to base URI

### ISSUE THREAD

**Issue by Ndiritu** [microsoftgraph/msgraph-sdk-php-core#195]
### Is your feature request related to a problem? Please describe the problem.

Currently, the base URI only contains the national cloud host, meaning each request needs a Graph API version prefix:
```php
$httpClient = GraphClientFactory::createWithConfig($config);
$response = $httpClient->get("/v1.0/me");
```

Current workaround is to override the `base_uri` via config:
```php

$httpClient = GraphClientFactory::createWithConfig(['base_uri' => NationalCloud::Global + "/v1.0"]);
$response = $httpClient->get("/me");
```

This would be a breaking change.

### Describe the solution you'd like.

Support setting the Graph API version via the factory & default to `v1.0`
(With default v1.0)
```php

$httpClient = GraphClientFactory::create();
$response = $httpClient->get("/me");
```

Or to change to `beta`
```php

$httpClient = GraphClientFactory::setApiVersion('beta')::create();
$response = $httpClient->get("/me");
```

**Comment by andrueastman** [comment]
This does not necessarily have to be a breaking change. 
We could always add an overload that takes in version parameter or add a property to the config object. 

https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/blob/9ed307008545e19b516e35e57ff35df061eaf05d/src/Microsoft.Graph.Core/Requests/GraphClientFactory.cs#L57

**Comment by Ndiritu** [comment]
Good catch, yes we can add a version parameter that defaults to an empty string. 
The config object ideally maps to a [Guzzle request options](https://docs.guzzlephp.org/en/stable/request-options.html) object.
